mailbox.hpp

```diff
@@ -86,7 +86,7 @@ void mailbox<T>::queue (int key, buffer_ptr<T> ptr) {
 
     } else {
         box.emplace(key, std::make_shared<map_value>(max_read, ptr));
-        box[key]->val_cv.notify_one();
+        box[key]->val_cv.notify_all();
```

Comment:
put this in a separate commit?

Reply:
Yea that should probably be in the other PR